apis/apiextensions/v1/composition_environment.go

```diff
@@ -194,6 +194,10 @@ type EnvironmentSourceSelector struct {
 	// MaxMatch specifies the number of extracted EnvironmentConfigs in Multiple mode, extracts all if nil.
 	MaxMatch *uint64 `json:"maxMatch,omitempty"`
 
+	// MinMatch specifies the required minimum of extracted EnvironmentConfigs in Multiple mode.
+	// +kubebuilder:default=1
+	MinMatch uint64 `json:"minMatch,omitempty"`
```

Comment:
```suggestion
	MinMatch *uint64 `json:"minMatch,omitempty"`
```
Given that `minMatch` is not meant to be set in Single mode, would it be possible to keep it a pointer too, remove the default and expand the check we do for MaxMatch in EnvironmentSourceSelector.Validate method to also enforce this not being set in Single mode?

We do already have a few places where `nil` is handled as `1`, but in this case I'd make it mean `0`, just to avoid introducing a breaking change also for `Multiple` mode.

I think it's fine to have `Single` mode meaning "exactly" one match, while `Multiple` defaults to any number of matches, but can be constrained further if needed.